Gradata/tests/test_cloud_events_pull.py

```
@@ -0,0 +1,196 @@
+
+        def read(self):
+            return self._data
+
+        def __enter__(self):
+            return self
+
+        def __exit__(self, *a):
+            return False
+
+    with patch("urllib.request.urlopen", return_value=FakeResp(body)):
+        with pytest.raises(NotImplementedError, match="Phase 2"):
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)